luavm/src/lib/instructions/loads.rs

```diff
@@ -1,28 +1,24 @@
 use lua_value::*;
 use luacompiler::bytecode::instructions::{first_arg, second_arg};
 use Vm;
-use VmChange;
 
-pub fn mov(_vm: &Vm, instr: u32) -> VmChange {
+pub fn mov(vm: &mut Vm, instr: u32) {
     let i = first_arg(instr) as usize;
     let j = second_arg(instr) as usize;
-    VmChange::RegSwap(i, j)
+    vm.registers.swap(i, j);
```

Comment:
Moving isn't the same as swapping... or am I missing out on something?

Reply:
It isn't, but I *think* this might faster than moving the `Box` from `j` to `i` and then initializing a new `Box` at index `j`.

Reply:
This avoids the extra heap allocation.

Reply:
But isn't it semantically incorrect? "MOV A, B" means A and B should have the same contents; but "SWAP A, B" means they have different contents. Thus if I refer to A *and* B afterwards (at the very least), I'm going to get the wrong values.

Reply:
Let me fix that, you are right!